hack/generator/pkg/codegen/pipeline/add_cross_resource_references.go

```diff
@@ -18,7 +18,7 @@ import (
 	"github.com/Azure/azure-service-operator/hack/generator/pkg/config"
 )
 
-var armIDDescriptionRegex = regexp.MustCompile("(?i).*/subscriptions/.*?/resourceGroups/.*")
+var armIDDescriptionRegex = regexp.MustCompile("(?i)(.*/subscriptions/.*?/resourceGroups/.*|ARM ID|Resource ID)")
```

Comment:
This regex looks like it is matching a URL - why does it have spaces in it?

Reply:
This regex is basically a heurisitc looking through the property comment to guess if it is an ARM ID. It has a URL in it not because the comment must contain a URL, but because many comments for ARM IDs say: "This is of the form: /subscriptions/<>/resourceGroups/<>/microsoft.storage/..."

The other two fields I added are additional detection that are common - the comments often say "ARM ID" or "Resource ID"

This is really a subset of the heuristics I came up with for the Swagger repo linter (which hasn't been merged yet).